### PR TITLE
General cleanup and add RTIC example

### PR DESCRIPTION
**Pull request by jonapap**

### CODE REVIEW

**Comment by NoahSprenger** [comment]
Looks good! For the logging script can we add another parameter for the port since my VSCode debugging setup takes port 60000.

**Review by NoahSprenger — DISMISSED** [review]

File: libraries/common-arm/memory.x [hunk 1]
```diff
@@ -0,0 +1,7 @@
+MEMORY
+{
+  /* NOTE K = KiBi = 1024 bytes */
+  FLASH (rx) : ORIGIN = 0x00000000, LENGTH = 256K
```
Comment (jonapap) [inline libraries/common-arm/memory.x:4]: Now using the memory size for the ATSAME51J18 (the one we'll actually use on the rocket)

Comment (Text-Input) [reply libraries/common-arm/memory.x:4]: Does the memory layout defined here in this 'common-arm' library get used by all boards that define this package as a dependency? Trying to figure out what this package is being used for.

Comment (jonapap) [reply libraries/common-arm/memory.x:4]: Yes, this file just needs to be present in the dependency tree. Also, my idea is that `common-arm` will eventually become the main crate, containing any non-board specific code. I've started to use it a bit more in the error manager PR.

Comment (Text-Input) [reply libraries/common-arm/memory.x:4]: Ah, that makes more sense. Thanks!

File: boards/main/Cargo.toml [hunk 2]
```diff
@@ -15,6 +15,4 @@ sdio-host = "0.9.0"
 defmt-rtt = "0.4.0"
 defmt = "0.3.2"
 atsamd-hal = { workspace = true }
-#
-#[package.metadata.docs.rs]
-#targets = ["thumbv7em-none-eabihf"]
```
Comment (Text-Input) [inline boards/main/Cargo.toml:20]: I can't see anywhere else that this is added and I think we still want it for local documentation, can you maybe add this line to 'common-arm'?

Comment (jonapap) [reply boards/main/Cargo.toml:20]: Do we need it? In my experience, the docs are building fine without this line

Comment (Text-Input) [reply boards/main/Cargo.toml:20]: Just built the docs locally, looks like you're correct. I'm fine with removing it then.

**Review by Text-Input — APPROVED** [review]
Looks good to me!

**Pull request merged**